src/data/FindIndex.js

```
@@ -26,37 +26,47 @@
+    #modelConstructor;
 
     /**
      * Constructs a new `Query` instance with the provided query object.
      *
-     * @param {Object} query - The structured query object defining the search criteria.
+     * @param {Model.constructor} constructor - The model class.
+     * @param {Record<string, Model>} index - The index dataset to search through.
      */
-    constructor(query) {
-        this.query = query;
+    constructor(constructor, index) {
```

Biome (1.9.4)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)